core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/PythonProxyServer.scala

```diff
@@ -63,10 +66,20 @@ private class AmberProducer(
 
     val root = flightStream.getRoot
 
+    // send back ack with credits on ackStream
+    val bufferAllocator = new RootAllocator(8 * 1024)
+    try {
+      val arrowBuf: ArrowBuf = bufferAllocator.buffer(4 * 1024)
```

Comment:
if we are sending a Long value, do we really need this large of a buffer (4*1024 bytes)?

Reply:
Updated it to use 12 bytes for longs, thank you!